packages/web-shared/hooks/useLink.js

```diff
@@ -0,0 +1,42 @@
+import { useCallback } from 'react';
+import { isValidUrl } from '../utils';
+import useCurrentUser from './useCurrentUser';
+
+const ROCK_APP_HOST_REGEX = /rock\.[a-z]+\.app/i;
+
+const useLink = () => {
+  const { currentUser } = useCurrentUser();
+  const rockAuthToken = currentUser?.rock?.authToken;
+
+  const transformLink = useCallback(
+    (link, options) => {
+      const { useRockAuth = true } = options || {};
+      // isValidUrl also uses URL under the hood.
+      // this should guarantee the destructuring
+      // below won't fail.
+      if (!isValidUrl(link)) return link;
+
+      const tokenizedUrl = new URL(link);
+
+      const { host, protocol, searchParams } = tokenizedUrl;
+
+      if (protocol !== 'https:') return link;
+
+      // `<a href="https://rock.apollos.app/MyAccount">rock link</a>`
+      const appendRockAuthToken = useRockAuth && rockAuthToken && ROCK_APP_HOST_REGEX.test(host);
```

Comment:
your best bet here is to pull in the `ROCK.API` from the config for that church, and check if the url you are inspecting matches the hostname of the `rock.api` url. 

@jordanwade may have to provide feedback on how to best pull in the `rock.api` value

Reply:
@vinnyjth is it ok if we expose `ROCK.API` on the `currentChurch` query?
https://github.com/ApollosProject/apollos-cluster/blob/43e38b699fd9dcf95576754a741ca4795e6488ad/src/data/church/resolver.js#L7

Reply:
```
rockUrl: async ({ slug }) => {
      const configValue = await sequelize.models.config.findOne({
        where: { churchSlug: slug, key: 'ROCK.API' },
      });
      return configValue?.value;
    },

Reply:
yah! but might need to make it more generalized. at the risk of overingeering

```
type ChurchDomain {
  type: String
  url: String
}

extend type Church {
  domains: [ChurchDomain]
}
```

We'll have a couple other authed church urls down the line, and I want to build with those integrations in mind.

Reply:
Upon further reflection @jordanwade @lefleraa - what do we think about using the current website hostname for now? 

Essentially if I'm on `cedarcreek.com` and  the url in the message is `rock.cedarcreek.com` - add the auth token. Rather than thinking about it like "this url is a rock url", think about it like "this url is owned by the church"

Reply:
I don’t think I follow…

Reply:
Initially this scope was for just Rock urls, then I assumed your suggestion was to make it work for any url.

Now I’m lost with this last suggestion. Maybe @lefleraa is following better than me.

Reply:
@vinnyjth I am with ya Jordan, I think. But if I am reading it correctly, that is what the regex is trying to do, anything `rock.*.app` appends the token. @redreceipt had a comment saying that it is not always the format, though? Saying sometimes the url is simply something like `newspring.cc`.

Reply:
But I think you are asking for one more step to make sure that the `*` contains the hostname?

Reply:
@vinnyjth bump ^